### PR TITLE
refactor: remove useSafeAreaInsets() | export header height

### PR DESCRIPTION
**Pull request by rajsam003**
This PR 

- removes the usage of `useSafeAreaInsets()` wherever possible and adds only to the header component. 
- Exports header height to use on different files

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

### CODE REVIEW

File: src/app/screens/HomeTabs/Settings/DarkModeSettings.tsx [hunk 1]
```diff
@@ -10,35 +10,36 @@ export const DarkModeSettings = () => {
   const syncWithSystem = true
   const forceMode = "dark"
   return (
-    <Flex m={2} mt={6}>
-      <BackButton />
-      <Text variant="lg">Dark Mode Settings</Text>
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Settings/DarkModeSettings.tsx:15]: Lets use shared/UI Header everywhere

File: src/app/screens/HomeTabs/HomeTabs.tsx [hunk 2]
```diff
@@ -9,14 +9,16 @@ import { GlobalStore } from "app/store/GlobalStore"
...
 export const Header = () => {
   const navigation = useNavigation()
+  const safeAreaInsets = useSafeAreaInsets()
 
   return (
-    <Flex flexDirection="row" px={2} mt={2} alignItems="center" justifyContent="space-between">
+    <Flex flexDirection="row" px={2} mt={2} alignItems="center" pt={safeAreaInsets.top}>
       <Touchable
         onPress={() => {
           navigation.navigate("Settings")
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/HomeTabs.tsx:24]: We need to fix this type issue as well.

File: src/app/sharedUI/Header.tsx [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
 }
 
 export const Header = ({ label, rightElements }: HeaderProps) => {
+  const navigation = useNavigation()
+  const safeAreaInsets = useSafeAreaInsets()
   return (
-    <Flex px={2}>
-      <BackButton />
+    <Flex px={2} pt={safeAreaInsets.top}>
+      <Flex height={HEADER_HEIGHT} justifyContent="center">
```
Comment (pvinis) [inline src/app/sharedUI/Header.tsx:17]: these two should be combined. we should not have anything wrapping the Flex with the `height` prop.

File: src/app/wrappers/TabsContainer.tsx [hunk 4]
```diff
@@ -15,12 +14,11 @@ export const TabsContainer: React.FC<TabsContainerProps> = ({
   children,
   initialTabName,
 }) => {
-  const insets = useSafeAreaInsets()
   const space = useSpace()
 
   return (
     <>
-      <Flex flex={1} pt={insets.top}>
+      <>
```
Comment (pvinis) [inline src/app/wrappers/TabsContainer.tsx:21]: good stuff!

**Review by pvinis — CHANGES_REQUESTED** [review]

File: src/app/sharedUI/Header.tsx (hunk 3 above)
Comment (rajsam003) [reply src/app/sharedUI/Header.tsx:17]: Oh yes. Good catch.

Comment (rajsam003) [reply src/app/sharedUI/Header.tsx:17]: But we have header height 44 only for the back button, not for the label. So If we combine both flexes then the label flex gets messed up.  
On the artwork screen, we just import the back button height as the header height, as the screen doesn't contain any label/title. So I think it's better this way. 
Does it make sense?

File: src/app/wrappers/TabsContainer.tsx (hunk 4 above)
Comment (patrinoua) [inline src/app/wrappers/TabsContainer.tsx:21]: is this <> really necessary?

Comment (pvinis) [reply src/app/wrappers/TabsContainer.tsx:21]: haha interesting. in that specific case here, no. i commented on the rest of them, but i guess i picked the one that actually doesn't need to be there 😅

Comment (patrinoua) [reply src/app/wrappers/TabsContainer.tsx:21]: hihi 😄 so I guess we can remove this and merge, it looks good!

Comment (pvinis) [reply src/app/wrappers/TabsContainer.tsx:21]: we will pair with sam now for a bit. we can remove this, and we need to check some flex combinations

**Pull request merged**